storm-core/src/jvm/backtype/storm/blobstore/SyncBlobs.java

```diff
@@ -0,0 +1,132 @@
+ */
+package backtype.storm.blobstore;
+
+import backtype.storm.nimbus.NimbusInfo;
+import backtype.storm.utils.Utils;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.thrift.transport.TTransportException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Is called periodically and updates the nimbus with blobs based on the state stored inside the zookeeper
+ */
+public class SyncBlobs {
+  private static final Logger LOG = LoggerFactory.getLogger(SyncBlobs.class);
+  private CuratorFramework zkClient;
+  private Map conf;
+  private BlobStore blobStore;
+  private List<String> blobStoreKeyList = new ArrayList<String>();
+  private List<String> zookeeperKeyList = new ArrayList<String>();
+  private NimbusInfo nimbusInfo;
+
+  public SyncBlobs(BlobStore blobStore, Map conf) {
+    this.blobStore = blobStore;
+    this.conf = conf;
+  }
+
+  public void setNimbusInfo(NimbusInfo nimbusInfo) {
+    this.nimbusInfo = nimbusInfo;
+  }
+
+  public void setZookeeperKeyList(List<String> zookeeperKeyList) {
+    this.zookeeperKeyList = zookeeperKeyList;
+  }
+
+  public void setBlobStoreKeyList(List<String> blobStoreKeyList) {
+    this.blobStoreKeyList = blobStoreKeyList;
+  }
+
+  public NimbusInfo getNimbusInfo() {
+    return nimbusInfo;
+  }
+
+  public List<String> getBlobStoreKeyList() {
+    List<String> keyList = new ArrayList<String>();
+    keyList.addAll(blobStoreKeyList);
+    return keyList;
+  }
+
+  public List<String> getZookeeperKeyList() {
+    List<String> keyList = new ArrayList<String>();
+    keyList.addAll(zookeeperKeyList);
+    return keyList;
+  }
+
+  public synchronized void syncBlobs() {
+    try {
+    LOG.debug("Sync blobs - blobstore {} keys {} zookeeperkeys {}", blobStore, getBlobStoreKeyList(), getZookeeperKeyList());
+    zkClient = Utils.createZKClient(conf);
+    deleteKeyListFromBlobStoreNotOnZookeeper(getBlobStoreKeyList(), getZookeeperKeyList());
+    updateKeyListForBlobStore(getBlobStoreKeyList());
+    List<String> keyListToDownload = getKeyListToDownload(getBlobStoreKeyList(), getZookeeperKeyList());
+    LOG.debug("Key List Blobstore-> Zookeeper-> DownloadList {}-> {}-> {}", getBlobStoreKeyList(), getZookeeperKeyList(), keyListToDownload);
+
+    for (String key : keyListToDownload) {
+      List<NimbusInfo> nimbusInfoList = Utils.getNimbodesWithLatestVersionOfBlob(zkClient, key);
+      if(Utils.downloadMissingBlob(conf, blobStore, key, nimbusInfoList)) {
+        Utils.createStateInZookeeper(conf, key, nimbusInfo);
+      }
+    }
+    if (zkClient !=null) {
+      zkClient.close();
+    }
+    } catch(InterruptedException exp) {
+        LOG.error("InterruptedException {}", exp);
+    } catch(TTransportException exp) {
+        throw new RuntimeException(exp);
+    } catch(Exception exp) {
+      // Should we log or throw exception
+        throw new RuntimeException(exp);
```

Comment:
Let's just throw RTE here, unless there is something special we can do here for this Exception?